plugins/curation/server/__init__.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2016 Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+from girder.api import access
+from girder.api.describe import Description, describeRoute
+from girder.api.rest import Resource, loadmodel
+from girder.constants import AccessType
+from girder.utility import mail_utils
+import datetime
+import posixpath
+
+
+CURATION = 'curation'
+
+ENABLED = 'enabled'
+STATUS = 'status'
+TIMELINE = 'timeline'
+ENABLE_USER_ID = 'enableUserId'
+REQUEST_USER_ID = 'requestUserId'
+REVIEW_USER_ID = 'reviewUserId'
+
+CONSTRUCTION = 'construction'
+REQUESTED = 'requested'
+APPROVED = 'approved'
+
+DEFAULTS = {
+    ENABLED: False,
+    STATUS: CONSTRUCTION,
+}
+
+
+class CuratedFolder(Resource):
+
+    @access.public
+    @loadmodel(model='folder', level=AccessType.READ)
+    @describeRoute(
+        Description('Get curation details for the folder.')
+        .param('id', 'The folder ID', paramType='path')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Read permission denied on the folder.', 403)
+    )
+    def getCuration(self, folder, params):
+        result = dict(DEFAULTS)
+        result[TIMELINE] = []
+        result.update(folder.get(CURATION, {}))
+        return result
+
+    @access.public
+    @loadmodel(model='folder', level=AccessType.READ)
+    @describeRoute(
+        Description('Set curation details for the folder.')
+        .param('id', 'The folder ID', paramType='path')
+        .param('enabled', 'Enable or disable folder curation.',
+               required=False, dataType='boolean')
+        .param('status', 'Set the folder curation status.',
+               required=False, dataType='string')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Write permission denied on the folder.', 403)
+    )
+    def setCuration(self, folder, params):
```

Comment:
A few other things to note that might be helpful:

* The `describeRoute` decorator generates the swagger, but will not actually validate the params
* The `id` param will be validated through the `loadModel` decorator, against the `level` param to that decorator, to ensure that the folder with that `id` exists and that the current user has the `level` of access
* For required params, you can use [requireParams](https://github.com/girder/girder/blob/e46ff18eb0d0b3c34ca6432c188ba5cdbe477f40/plugins/thumbnails/server/rest.py#L55)
* You can use an [enum](https://github.com/girder/girder/blob/e46ff18eb0d0b3c34ca6432c188ba5cdbe477f40/plugins/thumbnails/server/rest.py#L47-L48) as a param type in the `describeRoute` decorator, but you would need to [enforce](https://github.com/girder/girder/blob/e46ff18eb0d0b3c34ca6432c188ba5cdbe477f40/plugins/thumbnails/server/rest.py#L61-L64) it yourself